src/components/collect-button.tsx

```diff
@@ -55,13 +65,36 @@ export class CollectButtonComponent extends BaseComponent<IProps, IState> {
   }
   private addTestMouseToBackpack = () => {
     const {backpack} = this.stores;
+    const {backpackIndex} = this.props;
+    if (backpack.getMouseAtIndex(backpackIndex)) {
+      return;
+    }
+
     const randSex = Math.random() > .5 ? "male" : "female";
     const randGenotype = Math.random() > .5 ? (Math.random() > .5 ? "BB" : "Bb") :
                                                   (Math.random() > .5 ? "bB" : "bb");
     backpack.addCollectedMouse(Mouse.create({sex: randSex, genotype: randGenotype, label: "lbl"}));
   }
   private handleClickRemove = () => {
     const {backpack} = this.stores;
-    backpack.removeCollectedMouse(this.props.index);
+    const {backpackIndex} = this.props;
+    if (backpackIndex != null) {
+      backpack.removeCollectedMouse(backpackIndex);
+    }
+  }
+
+  private handleClickMouse = () => {
+    const { backpack } = this.stores;
+    const { backpackIndex } = this.props;
+    if (backpack.activeSlot === backpackIndex) {
```

Comment:
This could also be handled in the model logic. The view could just call `backpack.toggleSelection(backpackIndex)`.

Reply:
Made this change, much cleaner!